HW2/P4/driver.py

```diff
@@ -15,14 +15,23 @@
 from timer import Timer
 import threading
 
-def py_median_3x3(image, iterations=10, num_threads=1):
+def py_median_3x3(image, iterations=10, num_threads = 4):
     ''' repeatedly filter with a 3x3 median '''
     tmpA = image.copy()
     tmpB = np.empty_like(tmpA)
 
     for i in range(iterations):
-        filtering.median_3x3(tmpA, tmpB, 0, 1)
+        # filtering.median_3x3(tmpA, tmpB, 0, 1)
         # swap direction of filtering
+        thread_list = []
+        for i in range(num_threads):
+            thread = threading.Thread(target = filtering.median_3x3, args = (tmpA, tmpB, i, num_threads))  
+            thread.start()
+            thread_list.append(thread)
+
+        for j in thread_list:
+            j.join()
```

Comment:
You could use an event for each thread to make this faster, so that threads don't have to wait for all other threads to finish an iteration.